### ISSUE TITLE
Hotkey options for direct save/load states

### ISSUE THREAD

**Issue by kimimaru4000** [libretro/RetroArch#12990]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

This is a follow-up to https://github.com/libretro/RetroArch/issues/12144. I would like to discuss the possibility of adding hotkeys for direct save and load states into RetroArch. I have added it to my own fork, and you can see what it l looks like in the image below:

![RetroArch_DirectSaveLoadHotkeys_640](https://user-images.githubusercontent.com/1388037/133341224-2574c458-5a0c-4ae2-8bd1-b7258ced7b71.png)

Reasons I want to discuss this feature:

- As mentioned in the other issue, I run a Twitch Plays channel, and separate slots make it much easier for controlling permissions for savestates.
- Some people, including myself, prefer direct states as opposed to changing save slots then saving or loading. Other emulators have this feature as well.
- What I have now has only 6 slots present at all times. It may be better if the number of available slots were user-defined to a certain limit - say 10. There could be a menu option to enable direct saving/loading, and the menu options for each slot could disappear if this option were not enabled.

I am fully willing to work on a PR for this, considering I have already done a large chunk of the work implementing it.

**Comment by i30817** [comment]
Sounds good, the dolphin way is cool.

You probably don't want these set up by default though. RA hotkey menu is a mess and you can't even turn them to nil without going into the config file (this is _problem_ imo), so position you chose in a keyboard or controller by default would conflict with others, or in the case of the computer emulation cores conflict with text input (1-0 keys particularly but even the F-keys are not free of this).

**Comment by i30817** [comment]
In short i kind of suggest that if you implement this you also implement a way to finally turn hotkeys to nil/empty from the hotkey config menu, which for some inexplicable reason isn't possible in RA. If you add a reset option even better.

**Comment by kimimaru4000** [comment]
I agree about keeping them unmapped by default. There are only so many keys/buttons and this is a new feature that users shouldn't expect to have mapped, considering the existing savestate system has been in place for so long.

RetroArch already supports unmapping inputs with `delete` key on desktop (unsure about consoles and mobile). Go to the action you want on the menu and simply press `delete`, without selecting it to map it. Even if it weren't implemented, that should be a separate issue.

**Comment by i30817** [comment]
Yes, i was confused at first (it's not delete for me but 'a' as in the controller), but i wasn't thinking about only resetting. I was thinking about resetting it to default.

**Comment by kimimaru4000** [comment]
Sounds good! I will work on a PR with the additional options.
